nuxeo-tree-snapshot-core/src/main/java/org/nuxeo/snapshot/SnapshotableAdapter.java

```diff
@@ -132,19 +137,38 @@ public Snapshot createSnapshot(VersioningOption option) {
 
         DocumentModelList children = doc.getCoreSession().getChildren(doc.getRef());
 
-        String[] vuuids = new String[children.size()];
-
-        for (int i = 0; i < children.size(); i++) {
-            DocumentModel child = children.get(i);
-            if (!child.isFolder()) {
-                DocumentRef leafRef = createLeafVersion(child, option);
-                vuuids[i] = leafRef.toString();
+        List<DocumentModel> folders = new ArrayList<>();
+        List<DocumentModel> leafs = new ArrayList<>();
+        doc.getCoreSession().save();
+        IterableQueryResult res = doc.getCoreSession().queryAndFetch(
+                "SELECT ecm:uuid FROM Document WHERE ecm:parentId = '" + doc.getId() + "'",
```

Comment:
This must also include `AND ecm:currentLifeCycleState != 'deleted'` to be equivalent to `getChildren`.
Also for the orderable case you need `ORDER BY ecm:pos`.